src/test/validation_chainstatemanager_tests.cpp

```diff
@@ -15,15 +18,62 @@
 
 #include <boost/test/unit_test.hpp>
 
-BOOST_FIXTURE_TEST_SUITE(validation_chainstatemanager_tests, TestingSetup)
+struct ChainTestingSetup : public BasicTestingSetup {
+    boost::thread_group threadGroup;
+
+    explicit ChainTestingSetup(const std::string& chainName = CBaseChainParams::MAIN, const std::vector<const char*>& extra_args = {})
+        : BasicTestingSetup(chainName, extra_args)
+    {
+        // Ideally we'd move all the RPC tests to the functional testing framework
+        // instead of unit tests, but for now we need these here.
+        RegisterAllCoreRPCCommands(tableRPC);
+
+        m_node.scheduler = MakeUnique<CScheduler>();
```

Comment:
I'd suggest moving this below the "We have to run..." comment, since it fits with the code there.